test/test_matching_with_dont_cares.py

```diff
@@ -0,0 +1,20 @@
+import unittest
+
+from approximate_string_matching.matching_with_dont_cares import\
+  exact_matching_with_dont_cares
+
+class ExactMatchingWithDontCaresCase(unittest.TestCase):
```

Comment:
Please add larger random and smaller exhaustive tests, like e.g. in test_string_indexing.TestSuffixArrays class.

Reply:
Added method to already existing test_approximate_string_matching which does exactly that.